src/avm2/native.ts

```diff
@@ -1063,14 +1063,17 @@ module Shumway.AVM2.AS {
       if (re === (void 0) || re === null) {
         return null;
       } else {
-        if (re instanceof XRegExp && re.global) {
-          var matches = [], m;
-          while ((m = re.exec(this))) {
-            matches.push(m[0]);
+        if (re instanceof ASRegExp) {
```

Comment:
Because of the way we initialize AS classes, you have to use `AS.ASRegExp` here. Otherwise, it won't find the correct constructor. This will change in the future, but for now it is what it is.